cmd/gateway-common.go

```diff
@@ -0,0 +1,325 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"net/http"
+
+	"github.com/minio/minio/pkg/errors"
+	"github.com/minio/minio/pkg/hash"
+
+	minio "github.com/minio/minio-go"
+)
+
+var (
+	// CanonicalizeETag provides canonicalizeETag function alias.
+	CanonicalizeETag = canonicalizeETag
+
+	// MustGetUUID function alias.
+	MustGetUUID = mustGetUUID
+
+	// ErrorIf provides errorIf function alias.
+	ErrorIf = errorIf
+
+	// FatalIf provides fatalIf function alias.
+	FatalIf = fatalIf
```

Comment:
this needs to be done better later. Maybe move ErrorIf and FatalIf to a common package

MustGetUUID and CanonicalizeETag can also be moved to a common package in future (or even now up to you)

Reply:
That is the next step.